frontend/ui/widget/inputtext.lua

```diff
@@ -141,8 +143,90 @@ if Device:isTouchDevice() or Device:hasDPad() then
                 self.text_widget:moveCursorToXY(x, y, true) -- restrict_to_view=true
                 self.charpos, self.top_line_num = self.text_widget:getCharPos()
             end
-            if Device:hasClipboard() and Device.input.hasClipboardText() then
-                self:addChars(Device.input.getClipboardText())
+            -- clipboard dialog
+            if Device:hasClipboard() then
+                if self.do_select then -- select mode on
+                    if self.selection_start_pos then -- select end
+                        Device.input.setClipboardText(table.concat(self.charlist, "", self.selection_start_pos, self.charpos-1))
```

Comment:
Might want to check how this behaves if I set the end of selection BEFORE the start i choose previously.
(And swap them if wrong order.)

Reply:
Thanks for an idea. It was safe (just empty clipboard), but I've inserted a check, they would be swapped if needed,
So we can select start and end in any order.